warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/IngestFieldFilter.java

```diff
@@ -47,42 +71,269 @@ public IngestFieldFilter(Type dataType) {
      * @param conf
      */
     public void setup(Configuration conf) {
-        Multimap<String,String> temp = TreeMultimap.create();
-        String fieldsStr = conf.get(dataType.typeName() + FILTER_FIELD_SUFFIX);
-        
-        if (StringUtils.isNotBlank(fieldsStr)) {
-            try {
-                for (String pair : fieldsStr.split(PAIR_DELIM)) {
-                    String[] tokens = pair.split(VALUE_DELIM);
-                    if (tokens.length == 2) {
-                        temp.put(tokens[0], tokens[1]);
-                    } else {
-                        logger.warn("Expected a " + VALUE_DELIM + " delimited pair but received: " + pair + ", ignoring this config.");
-                    }
-                }
-                
-            } catch (Exception e) {
-                logger.warn("Could not configure with the string: " + fieldsStr + ", Disabling field filtering.");
-            }
-        }
-        
-        filters = temp.asMap();
-        
-        logger.info("Filters for " + dataType.typeName() + ": " + filters);
+        fieldNameFilters = new FieldConfiguration();
+        fieldNameFilters.load(conf.get(dataType.typeName() + FILTER_FIELD_SUFFIX), false);
+        fieldNameFilters.load(conf.get(dataType.typeName() + FILTER_FIELD_NAME_SUFFIX), false);
+        logger.info("Field Name Filters for " + dataType.typeName() + ": " + fieldNameFilters);
+
+        fieldValueFilters = new FieldConfiguration(conf.get(dataType.typeName() + FILTER_FIELD_VALUE_SUFFIX), true);
+        logger.info("Field Value Filters for " + dataType.typeName() + ": " + fieldValueFilters);
     }
-    
+
     /**
      * Applies the configured filter rules to the given fields.
      *
      * @param fields
      */
     public void apply(Multimap<String,?> fields) {
-        for (Map.Entry<String,Collection<String>> entry : filters.entrySet()) {
-            if (fields.containsKey(entry.getKey())) {
-                for (String toRemove : entry.getValue()) {
-                    fields.removeAll(toRemove);
+        // for the field name filters, remove the drop fields if the keep fields exist
+        for (FieldFilter filter : fieldNameFilters) {
+            if (fields.keySet().containsAll(filter.getKeepFields())) {
+                for (String fieldToRemove : filter.getDropFields()) {
```

Comment:
No need to iterate if you're using removeAll.

Reply:
presuming you mean fields.getKeySet().removeAll(filter.getDropFields()).  The removeAll on a Multimap only support a single key.